### DISCUSSION TITLE
systemd default tmpfiles removes /tmp/.X11-unix/X0

### DISCUSSION THREAD

**Discussion by myme** [nix-community/NixOS-WSL#113] [Q&A] [answered]
I'm fairly certain I've been able to run `x11` apps on NixOS-WSL before, but suddenly it stopped working. Wayland apps still worked, so figured it was related to `xwayland` somehow. Found https://github.com/microsoft/wslg/issues/558 & https://randombytes.substack.com/p/wslg-on-making-wslg-and-systemd-play which seemed to describe my error case quite perfectly. `systemd-tmpfiles` is removing the `/tmp/.X11-unix` directory created by WSL and so the `X0` socket is gone.

Doing this in my config makes `x11` apps work again:

```nix
{
      systemd.tmpfiles.packages = lib.mkForce [];
}
```

I'm not sure if I'm actually relying on any of the `tmpfiles` being setup by `systemd`, but clearing the list seems a bit overkill in any case. I haven't been able to find any NixOS-WSL bugs or reports where people experience this. I'm seeing this on two different machines, so I'm curious as to why I'm not finding anybody else with this issue.

Is it a bug that should be handled through NixOS-WSL somehow or just me?

**Comment by nzbr** [comment, marked as answer]
I have created an issue (#114) for this
